tutorials/2d/2d_movement.rst

```diff
@@ -36,77 +36,52 @@ fact that the player can move diagonally by pressing two keys at the same time.
 
 .. image:: img/movement_8way.gif
 
-Add a script to the kinematic body and add the following code:
+Add a script to the character body and add the following code:
 
 .. tabs::
  .. code-tab:: gdscript GDScript
 
     extends CharacterBody2D
 
-    export (int) var speed = 200
-
-    var velocity = Vector2()
+    @export var speed = 400
 
     func get_input():
-        velocity = Vector2()
-        if Input.is_action_pressed("right"):
-            velocity.x += 1
-        if Input.is_action_pressed("left"):
-            velocity.x -= 1
-        if Input.is_action_pressed("down"):
-            velocity.y += 1
-        if Input.is_action_pressed("up"):
-            velocity.y -= 1
-        velocity = velocity.normalized() * speed
+        var input_direction = Input.get_vector("left", "right", "up", "down")
+        velocity = input_direction * speed
 
     func _physics_process(delta):
         get_input()
-        velocity = move_and_slide(velocity)
+        move_and_slide()
 
  .. code-tab:: csharp
 
     using Godot;
     using System;
 
-    public class Movement : CharacterBody2D
+    public partial class Movement : CharacterBody2D
     {
-        [Export] public int speed = 200;
-
-        public Vector2 velocity = new Vector2();
+        [Export]
+        public int Speed { get; set; } = 400;
 
         public void GetInput()
         {
-            velocity = new Vector2();
-
-            if (Input.IsActionPressed("right"))
-                velocity.x += 1;
-
-            if (Input.IsActionPressed("left"))
-                velocity.x -= 1;
 
-            if (Input.IsActionPressed("down"))
-                velocity.y += 1;
-
-            if (Input.IsActionPressed("up"))
-                velocity.y -= 1;
-
-            velocity = velocity.Normalized() * speed;
+            Vector2 inputDirection = Input.GetVector("left", "right", "up", "down");
+            Velocity = inputDirection * speed;
         }
 
-        public override void _PhysicsProcess(float delta)
+        public override void _PhysicsProcess(double delta)
         {
             GetInput();
-            velocity = MoveAndSlide(velocity);
+            MoveAndSlide();
         }
     }
 
-In the ``get_input()`` function, we check for the four key events and sum them
-up to get the velocity vector. This has the benefit of making two opposite keys
-cancel each other out, but will also result in diagonal movement being faster
-due to the two directions being added together.
+In the ``get_input()`` function, we use `Input.get_vector()` to check for the
```

Comment:
I think it may be useful to link to the `Input` documentation here, what do you think?

```suggestion
In the ``get_input()`` function, we use :ref:`Input <class_Input>` ``get_vector()`` to check for the
```

Otherwise, the syntax here should be using two single quotes.

```suggestion
In the ``get_input()`` function, we use ``Input.get_vector()`` to check for the
```